### ISSUE TITLE
Graph::Easy cyclical dependency with Carton and cpanfile.snapshot

### ISSUE THREAD

**Issue by ssoriche** [shlomif/Graph-Easy-As_svg#1]
When the distribution is built `Graph::Easy` is included in the meta data in the `provides` section in the file `lib/Graph/Easy/As_svg.pm`:
 
```
   "provides" : {
      "Graph::Easy" : {
         "file" : "lib/Graph/Easy/As_svg.pm",
         "version" : "0.26"
      },
```

`Graph::Easy` is also listed as a runtime requirement. 

```
      "runtime" : {
         "requires" : {
            "Graph::Easy" : "0.63",
```

Installation with `carton install` works fine, however when running `carton install --deployment`, carton tries to install `Graph::Easy`, defers it, tries to install `Graph::Easy::As_svg`, finds that it needs and provides `Graph::Easy` and that the installation is deferred, and fails to install.

Manually editing the `cpanfile.snapshot` to remove:

```
  Graph-Easy-As_svg-0.26
    pathname: S/SH/SHLOMIF/Graph-Easy-As_svg-0.26.tar.gz
    provides:
      Graph::Easy 0.26
```

`Graph::Easy` from the `provides` line solves the issue.

As `Graph::Easy::As_svg` isn't really providing `Graph::Easy` and requires the actual `Graph::Easy`, I think the proper solution is the removal of the provides.

**Comment by shlomif** [comment]
Hi!

Thanks for the report. Can you send me a pull-req that will fix this problem? I'll try asking the dzil ppl too.

**Comment by shlomif** [comment]
Fixed in 0.27. Thanks!
